File: freecodecamp/daily-code-challenge/2026-03-22.py
def detect_roast(beans):
    total = 0

    for ch in beans:
        if ch == "'":
            total += 1
        elif ch == "-":
            total += 2
        elif ch == ".":
            total += 3

    avg = total / len(beans)

    if avg < 1.75:
        return "Light"
    elif avg <= 2.5:
        return "Medium"
    else:
        return "Dark"

File: freecodecamp/daily-code-challenge/2026-03-22.js
function detectRoast(beans) {
  let total = 0;

  for (let ch of beans) {
    if (ch === "'") total += 1;
    else if (ch === "-") total += 2;
    else if (ch === ".") total += 3;
  }

  let avg = total / beans.length;

  if (avg < 1.75) return "Light";
  else if (avg <= 2.5) return "Medium";
  else return "Dark";
}